tests/common.go

```diff
@@ -10110,3 +10136,27 @@ func GetRandomSubset(elements []string, subsetSize int) ([]string, error) {
 
 	return shuffledElements[:subsetSize], nil
 }
+
+// DeleteAllNamespacesCreatedByTestCase deletes all the namespaces created for the test case
+func DeleteAllNamespacesCreatedByTestCase(CurrentTestRailTestCaseId int) error {
+
+	// Get all the namespaces on the cluster
+	k8sCore := core.Instance()
+	allNamespaces, err := k8sCore.ListNamespaces(make(map[string]string))
+	if err != nil {
+		return fmt.Errorf("error in listing namespaces. Err: %v", err.Error())
+	}
+
+	// Iterate and remove all namespaces
+	for _, ns := range allNamespaces.Items {
+		if strings.Contains(ns.Name, fmt.Sprintf("%d", CurrentTestRailTestCaseId)) {
+			log.Infof("Deleting namespace [%s]", ns.Name)
+			err = k8sCore.DeleteNamespace(ns.Name)
```

Comment:
the namespaces will have creator=torpedo label; it is probably best to use Inst().InstanceID

Reply:
I thought about this approach but after discussing with Mithun. We both felt this is a better approach as it will allow us to use the same code when we run tests in parallel.

Reply:
What about restored namespaces that do not have CurrentTestRailTestCaseId (or Inst().InstanceID)? There is no need to delete them?

Reply:
There are just 2-3 such cases the patch will clean up for the majority which will eventually reduce the number of test failures with the mentioned error.

Reply:
@kphalgun-px One of the reasons why we did not go with the label `creator=torpedo`, in future if we are running these tests on a cluster which has pre-existing namespaces created by torpedo which we do not want to delete (say in a BYOC job), then this approach is not suitable.

Reply:
pre-existing namespaces might have 'CurrentTestRailTestCaseId', right? I feel it is safe to to use Inst().InstanceID, which, I think, includes a timestamp

Reply:
@kshithijiyer-px After discussing with @kphalgun-px, I think it better we use `Inst().InstanceID` instead of `CurrentTestRailTestCaseId`
There are 2 advantages:

- When we enable parallel execution in future, `CurrentTestRailTestCaseId` will be a global variable which can cause deletion of incorrect namespace
- If we use `Inst().InstanceID` we dont have to pass any input to the function `DeleteAllNamespacesCreatedByTestCase`

Reply:
Okay done